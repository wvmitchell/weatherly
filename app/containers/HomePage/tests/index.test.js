import React from 'react';
import { shallow, mount } from 'enzyme';

import { HomePage, mapDispatchToProps } from '../index';

describe('<HomePage />', () => {
  let dispatch = jest.fn();

  it('Expect it to match snapshot', () => {
    const renderedComponent = shallow(<HomePage dispatch={dispatch} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('getTempExtremesString parses the forecast to produce the correct string', () => {
    const mockForecast = {
      simple: {
        forecastday: [
          {
            high: {
              fahrenheit: 88.2,
              celsius: 32.2,
            },
            low: {
              fahrenheit: 18.2,
              celsius: 2.2,
            },
          },
        ],
      },
    };
    const renderedComponent = mount(
      <HomePage
        dispatch={dispatch}
        submitLocation={jest.fn()}
      />
    );
    expect(renderedComponent.instance().getTempExtremesString(mockForecast, 0)).toEqual('High: 88.2 F (32.2 C) / Low: 18.2 F (2.2 C)');
  });

  it('searchForLocation submits the location search', () => {
    const clearConditions = jest.fn();
    const clearForecast = jest.fn();
    const submitLocation = jest.fn();
    const renderedComponent = mount(
      <HomePage
        dispatch={dispatch}
        clearConditions={clearConditions}
        clearForecast={clearForecast}
        submitLocation={submitLocation}
      />
    );
    renderedComponent.instance().searchForLocation({ preventDefault: jest.fn() });
    expect(clearConditions).toHaveBeenCalled();
    expect(clearForecast).toHaveBeenCalled();
    expect(submitLocation).toHaveBeenCalled();
  });

  it('selectSuggestion selects a location and updates the store', () => {
    const clearQueryResults = jest.fn();
    const clearConditions = jest.fn();
    const clearForecast = jest.fn();
    const submitLocation = jest.fn();
    const renderedComponent = mount(
      <HomePage
        dispatch={dispatch}
        clearQueryResults={clearQueryResults}
        clearConditions={clearConditions}
        clearForecast={clearForecast}
        submitLocation={submitLocation}
      />
    );
    renderedComponent.instance().selectSuggestion({ target: { value: 'some-location' } });
    expect(clearQueryResults).toHaveBeenCalled();
    expect(clearConditions).toHaveBeenCalled();
    expect(clearForecast).toHaveBeenCalled();
    expect(submitLocation).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    let result;

    beforeAll(() => {
      dispatch = jest.fn();
      result = mapDispatchToProps(dispatch);
    });

    it('injects dispatch', () => {
      expect(result.dispatch).toBeDefined();
    });

    it('defines addLocation fn', () => {
      result.addLocation({ target: { value: 'something' } });
      expect(dispatch).toHaveBeenCalled();
    });

    it('defines submitLocation fn', () => {
      result.submitLocation();
      expect(dispatch).toHaveBeenCalled();
    });

    it('defines clearQueryResults fn', () => {
      result.clearQueryResults();
      expect(dispatch).toHaveBeenCalled();
    });

    it('defines clearConditions fn', () => {
      result.clearConditions();
      expect(dispatch).toHaveBeenCalled();
    });

    it('defines clearForecast fn', () => {
      result.clearForecast();
      expect(dispatch).toHaveBeenCalled();
    });
  });
});
