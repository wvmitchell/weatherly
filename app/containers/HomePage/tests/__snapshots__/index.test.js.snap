exports[`<HomePage /> Expect it to match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <HelmetWrapper
        meta={
            Array [
                Object {
                  "content": "Find out what the weather is like in your area",
                  "name": "description",
                },
              ]
        }
        title="Weatherly - Weather happens" />
    <withStyles(AppBar)
        position="static">
        <withStyles(Toolbar)>
            <withStyles(Typography)
                color="inherit"
                type="title">
                Weatherly
            </withStyles(Typography)>
        </withStyles(Toolbar)>
    </withStyles(AppBar)>
    <withStyles(Typography)
        component="h1"
        type="display3">
        <FormattedMessage
            defaultMessage="What\'s the weather like where you are?"
            id="app.containers.HomePage.header"
            tagName="span"
            values={Object {}} />
    </withStyles(Typography)>
    <form
        onSubmit={[Function]}>
        <TextField
            label="Good question, let\'s see"
            onChange={undefined}
            required={false} />
        <withStyles(Button)
            color="accent"
            raised={true}
            type="submit">
            Go
        </withStyles(Button)>
    </form>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <HelmetWrapper
            meta={
                  Array [
                        Object {
                          "content": "Find out what the weather is like in your area",
                          "name": "description",
                        },
                      ]
            }
            title="Weatherly - Weather happens" />
      <withStyles(AppBar)
            position="static">
            <withStyles(Toolbar)>
                  <withStyles(Typography)
                        color="inherit"
                        type="title">
                        Weatherly
                  </withStyles(Typography)>
            </withStyles(Toolbar)>
      </withStyles(AppBar)>
      <withStyles(Typography)
            component="h1"
            type="display3">
            <FormattedMessage
                  defaultMessage="What\'s the weather like where you are?"
                  id="app.containers.HomePage.header"
                  tagName="span"
                  values={Object {}} />
      </withStyles(Typography)>
      <form
            onSubmit={[Function]}>
            <TextField
                  label="Good question, let\'s see"
                  onChange={undefined}
                  required={false} />
            <withStyles(Button)
                  color="accent"
                  raised={true}
                  type="submit">
                  Go
            </withStyles(Button)>
      </form>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <HomePage
        dispatch={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": HomePage {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "getTempExtremesString": [Function],
        "props": Object {
          "dispatch": [Function],
        },
        "refs": Object {},
        "selectSuggestion": [Function],
        "state": null,
        "submitLocation": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <HelmetWrapper
                    meta={
                              Array [
                                        Object {
                                          "content": "Find out what the weather is like in your area",
                                          "name": "description",
                                        },
                                      ]
                    }
                    title="Weatherly - Weather happens" />
          <withStyles(AppBar)
                    position="static">
                    <withStyles(Toolbar)>
                              <withStyles(Typography)
                                        color="inherit"
                                        type="title">
                                        Weatherly
                              </withStyles(Typography)>
                    </withStyles(Toolbar)>
          </withStyles(AppBar)>
          <withStyles(Typography)
                    component="h1"
                    type="display3">
                    <FormattedMessage
                              defaultMessage="What\'s the weather like where you are?"
                              id="app.containers.HomePage.header"
                              tagName="span"
                              values={Object {}} />
          </withStyles(Typography)>
          <form
                    onSubmit={[Function]}>
                    <TextField
                              label="Good question, let\'s see"
                              onChange={undefined}
                              required={false} />
                    <withStyles(Button)
                              color="accent"
                              raised={true}
                              type="submit">
                              Go
                    </withStyles(Button)>
          </form>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <HelmetWrapper
                    meta={
                              Array [
                                        Object {
                                          "content": "Find out what the weather is like in your area",
                                          "name": "description",
                                        },
                                      ]
                    }
                    title="Weatherly - Weather happens" />
          <withStyles(AppBar)
                    position="static">
                    <withStyles(Toolbar)>
                              <withStyles(Typography)
                                        color="inherit"
                                        type="title">
                                        Weatherly
                              </withStyles(Typography)>
                    </withStyles(Toolbar)>
          </withStyles(AppBar)>
          <withStyles(Typography)
                    component="h1"
                    type="display3">
                    <FormattedMessage
                              defaultMessage="What\'s the weather like where you are?"
                              id="app.containers.HomePage.header"
                              tagName="span"
                              values={Object {}} />
          </withStyles(Typography)>
          <form
                    onSubmit={[Function]}>
                    <TextField
                              label="Good question, let\'s see"
                              onChange={undefined}
                              required={false} />
                    <withStyles(Button)
                              color="accent"
                              raised={true}
                              type="submit">
                              Go
                    </withStyles(Button)>
          </form>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HomePage
    dispatch={[Function]} />,
}
`;
